src/game/file_io.c

```diff
@@ -314,8 +315,10 @@ static void init_scenario_data(scenario_version version)
         state->scenario = create_scenario_piece(1720, 0);
     } else if(version <= SCENARIO_LAST_NO_WAGE_LIMITS) {
         state->scenario = create_scenario_piece(1830, 0);
-    } else {
+    } else if(version <= SCENARIO_LAST_NO_EXTENDED_REQUESTS) {
```

Comment:
This is getting to a point where we're also starting to need a `scenario_version_data` struct with all piece sizes.

Reply:
Do you want me to add that as part of this PR?
If so, just give me a idea of what you expect for this.

Reply:
I think we can add this as-is, as well as the other PR, then we do a new PR with that feature.

Reply:
Looking at it now. Will give it a shot.
Basically make it save and load something like monument deliveries?

I.e. Write to buffer the byte size of a single request entry (determined by version), and the number of array entries (static 20 for now).
Then read that number of array elements when loading.
Means I also have to shuffle saving the request data to be in one place, and save an array element at a time.

Is that about right?

Reply:
Yes exactly. You probably don't need the total number of requests, as that will be "total buffer size - 4 bytes"/"budget size per request".